das/state.go

```diff
@@ -0,0 +1,54 @@
+package das
+
+import "sync/atomic"
+
+// State is an interface that allows other components to access the
+// current state of the DASer safely.
+type State interface {
+	IsCaughtUp() bool
+	LatestSampledHeight() uint64
+}
+
+// state collects information about the DASer process.
+type state struct {
+	// tracks whether sampling routine is running
+	isRunning uint64
+	// indicates whether the DASer catchUp routine
+	// has successfully caught up to network head.
+	caughtUp uint64
```

Comment:
This is used as a Boolean, so should be prefixed with `is` to make is clear from the name.
```suggestion
	isCaughtUp uint64
```

Reply:
done